### DISCUSSION TITLE
Paret-Front set

### DISCUSSION THREAD

**Discussion by Matty19941373** [anyoptimization/pymoo#377] [Q&A] [answered]
Hi All, 
I am learning the ropes in implementing Pymoo for my problem which is a multi-objective optimization. I have reviewed the Tutorial for Pymoo and could not find how to obtain a pareto front set of solutions for a problem with unknown pareto set. I believe that "res.F" and res.X" will return Non-dominated solutions. So, the question is how to get pareto-front? I need the pareto-front set to visualize and to plot the outputs.

Thanks in advance.

**Comment by bruscalia** [comment]
Hi, @Matty19941373 

I believe what you are looking for is exactly the results instance, of which the property F stores objective values, and X the decision variables. The definition of Pareto optimal solutions will usually be the same as non-dominated solutions.

**Reply by Matty19941373** [reply]
Thanks for the reply.
My purpose is just plotting a graph in which there are all solutions and then I can highlight the pareto-front set in that plot. I think res.F is the Non-dominated solutions, how can I have all possible solutions?

**Comment by blankjul** [comment, marked as answer]
Do you have a test problem? If yes, you can mathematically derive the Pareto set and Pareto Front. You can find some help in textbooks on how to do that. It can be challenging though depending on the functions. This is by the way what researchers have done for all test problems where we have a Pareto front available (in general but also in pymoo).

Otherwise, you are asking a fundamental question in optimization: Have I really found the optimum?
Because we can not answer this question, we are actually applying optimization in the first place.

**Reply by Matty19941373** [reply]
Thanks for the reply. No i have defined 4 objective functions which the pareto-front is not known for them. 
My purpose is just plotting a graph in which there are all solutions and then I can highlight the pareto-front set in that plot. I think res.F is the Non-dominated solutions, how can I have all possible solutions?

**Reply by blankjul** [reply]
As said you will have to derive the optimum. Unfortunately, there is no quick hack for this.
